packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js

```diff
@@ -4617,6 +4618,66 @@ QUnit.module('Aria accessibility', {
         fx.off = false;
     }
 }, () => {
+    QUnit.test('aria-label should be \'Read-only calendar\' and role = group when readOnly option is true', function(assert) {
+        const $element = this.$element;
+
+        $element.dxCalendar({
+            readOnly: true
+        }).dxCalendar('instance');
+
+        assert.strictEqual($element.attr('aria-label'), 'Read-only calendar', 'aria-label is set correctly');
+        assert.strictEqual($element.attr('role'), 'group', 'role is set correctly');
+    });
+
+    QUnit.test('aria-label and role should be removed when readOnly option is false', function(assert) {
+        const $element = this.$element;
+
+        $element.dxCalendar({
+            readOnly: false
+        }).dxCalendar('instance');
+        assert.notOk($element.attr('aria-label'), 'aria-label is not set');
+        assert.notOk($element.attr('role'), 'role is not set');
+    });
+
+    QUnit.test('aria-label and role should be removed when readOnly option is set to false on runtime', function(assert) {
+        const $element = this.$element;
+
+        const calendar = $element.dxCalendar({
+            readOnly: true
+        }).dxCalendar('instance');
+
+        calendar.option('readOnly', false);
+
+        assert.notOk($element.attr('aria-label'), 'aria-label is removed');
+        assert.notOk($element.attr('role'), 'role is removed');
+    });
+
+    QUnit.test('aria-label and role should be setted when readOnly option is set to true on runtime', function(assert) {
+        const $element = this.$element;
+
+        const calendar = $element.dxCalendar({
+            readOnly: false
+        }).dxCalendar('instance');
+
+        calendar.option('readOnly', true);
+
+        assert.strictEqual($element.attr('aria-label'), 'Read-only calendar', 'aria-label is set correctly');
+        assert.strictEqual($element.attr('role'), 'group', 'role is set correctly');
+    });
+
+    QUnit.test('aria-label attribute should be equal to custom localized text', function(assert) {
+        const localizedText = 'Read-only calendar';
```

Comment:
Heh, not it's the same as a default one :)
Kindly set another value here

Reply:
[6e4a276](https://github.com/DevExpress/DevExtreme/pull/27898/commits/6e4a276381dbf4ab16f3ad59c181c9a9d0f85b73)